docs/arch/adr-002-rook-distributed.md

```diff
@@ -0,0 +1,28 @@
+# ADR 2: Use Rook as provisioner for Multi Node kURL Installations
+
+The initial kURL installation specification defaulted to [Longhorn](https://longhorn.io/) as the volume provisioner for both Single and Multi node deployments. However, the experience with Longhorn has not lived up to our expectations. Namely, the instability, slow performance and lack of community support for Longhorn has forced us to switch to another storage provisioner. For Single node cluster deployments the default storage provisioner will be delivered by [OpenEBS](https://github.com/replicatedhq/kURL/blob/main/docs/arch/adr-001-use-openebs.md) while for Multi node clusters this role starts to be fullfilled by [Rook](https://rook.io/docs/rook/v1.10/Getting-Started/intro/).
+
+## Decision
+
+We will use [Rook](https://rook.io/docs/rook/v1.10/Getting-Started/intro/) as the default volume provisioner for Multi node Embedded clusters. Rook cluster tuning, namely number of `replicas` and `mons`, will be automatically managed by Ekco operator. Every available block device will be added to the Rook cluster through the `useAllDevices: true` option.
+
+## Status
+
+Accepted
+
+## Consequences
+
+- Ekco operator will have to behave as follow:
+    - On a fresh cluster install Ekco manages the Rook cluster by default
+    - The option `useAllDevices` will be enabled
+    - The number of `replicas` would be automatically set (up to a max of 3 replicas)
+    - Each new node in the cluster is automatically added to the Rook cluster
```

Comment:
Do all off these features need to be implemented in EKCO? Which ones are already there?

Reply:
Good point, I am clarifying this in a second version of this doc.